src/windows/common/svccommio.cpp

```diff
@@ -18,265 +18,165 @@ Module Name:
-    // a more elegant solution exists.
-    //
-    // N.B. Ignore failures setting the mode if the console has already
-    //      disconnected.
-    //
-
-    if (!SetConsoleMode(File, ConsoleMode))
-    {
-        switch (GetLastError())
-        {
-        case ERROR_INVALID_PARAMETER:
-            THROW_HR(WSL_E_CONSOLE);
-
-        case ERROR_PIPE_NOT_CONNECTED:
-            break;
 
-        default:
-            THROW_LAST_ERROR();
-        }
-    }
+bool IsConsoleHandle(_In_ HANDLE Handle)
+{
+    DWORD Mode;
+    return GetFileType(Handle) == FILE_TYPE_CHAR && GetConsoleMode(Handle, &Mode);
 }
 
-void ConfigureStdHandles(_Inout_ LXSS_STD_HANDLES_INFO& StdHandlesInfo)
+void TrySetConsoleMode(_In_ HANDLE Handle, _In_ DWORD Mode)
 {
-    //
-    // Check stdin to see if it is a console or another device. If it is
-    // a console, configure it to raw processing mode and VT-100 support. If the
-    // force console I/O is requested, ignore stdin and get active console input
-    // handle instead.
-    //
-
-    UINT NewConsoleInputCP = 0;
-    DWORD NewConsoleInputMode = 0;
-    BOOLEAN IsConsoleInput = StdHandlesInfo.IsConsoleInput;
-    BOOLEAN IsConsoleOutput = StdHandlesInfo.IsConsoleOutput;
-    BOOLEAN IsConsoleError = StdHandlesInfo.IsConsoleError;
-    DWORD SavedInputMode = StdHandlesInfo.SavedInputMode;
-    DWORD SavedOutputMode = StdHandlesInfo.SavedOutputMode;
-    UINT SavedInputCP = StdHandlesInfo.SavedInputCP;
-    UINT SavedOutputCP = StdHandlesInfo.SavedOutputCP;
-    CONSOLE_SCREEN_BUFFER_INFO ScreenBufferInfo;
-    auto RestoreInputHandle = wil::scope_exit_log(WI_DIAGNOSTICS_INFO, [&] {
-        if (NewConsoleInputCP != 0)
-        {
-            SetConsoleCP(SavedInputCP);
-        }
-
-        if (NewConsoleInputMode != 0)
-        {
-            ChangeConsoleMode(StdHandlesInfo.InputHandle, SavedInputMode);
-        }
-    });
-
-    IsConsoleInput = FALSE;
-    if ((GetFileType(StdHandlesInfo.InputHandle) == FILE_TYPE_CHAR) && (GetConsoleMode(StdHandlesInfo.InputHandle, &SavedInputMode)))
+    if (!SetConsoleMode(Handle, Mode))
     {
-        IsConsoleInput = TRUE;
-        NewConsoleInputMode = SavedInputMode;
-        WI_SetAllFlags(NewConsoleInputMode, (ENABLE_WINDOW_INPUT | ENABLE_VIRTUAL_TERMINAL_INPUT));
-        WI_ClearAllFlags(NewConsoleInputMode, (ENABLE_ECHO_INPUT | ENABLE_INSERT_MODE | ENABLE_LINE_INPUT | ENABLE_PROCESSED_INPUT));
-        ChangeConsoleMode(StdHandlesInfo.InputHandle, NewConsoleInputMode);
-
-        //
-        // Set the console input to the UTF-8 code page.
-        //
-
-        SavedInputCP = GetConsoleCP();
-        NewConsoleInputCP = CP_UTF8;
-        THROW_LAST_ERROR_IF(!::SetConsoleCP(NewConsoleInputCP));
-    }
-
-    bool RestoreMode = false;
-    bool RestoreCp = false;
-    auto RestoreOutput = wil::scope_exit([&] {
-        if (RestoreMode)
+        const auto Error = GetLastError();
+        if (Error != ERROR_INVALID_PARAMETER && Error != ERROR_PIPE_NOT_CONNECTED)
```

Comment:
Looks like currently throw `WSL_E_CONSOLE` if we hit this, what's the reason for the behavior change for that error code  ?